src/applications/vaos/appointment-list/components/RequestedAppointmentDetailsPage.jsx

```diff
@@ -0,0 +1,188 @@
+import React, { useEffect } from 'react';
+import { Link, useParams, useHistory } from 'react-router-dom';
+import { connect } from 'react-redux';
+import LoadingIndicator from '@department-of-veterans-affairs/component-library/LoadingIndicator';
+import moment from 'moment';
+
+import * as actions from '../redux/actions';
+import { FETCH_STATUS } from '../../utils/constants';
+import { lowerCase } from '../../utils/formatters';
+import { scrollAndFocus } from '../../utils/scrollAndFocus';
+import ListBestTimeToCall from './cards/pending/ListBestTimeToCall';
+import VAFacilityLocation from '../../components/VAFacilityLocation';
+
+import {
+  getPatientTelecom,
+  isVideoAppointment,
+  getVAAppointmentLocationId,
+} from '../../services/appointment';
+
+const TIME_TEXT = {
+  AM: 'in the morning',
+  PM: 'in the afternoon',
+  'No Time Selected': '',
+};
+
+// Only use this when we need to pass data that comes back from one of our
+// services files to one of the older api functions
+function parseFakeFHIRId(id) {
+  return id ? id.replace('var', '') : id;
+}
+
+function RequestedAppointmentDetailsPage({
+  appointmentDetails,
+  appointmentDetailsStatus,
+  facilityData,
+  fetchRequestDetails,
+  pendingStatus,
+  requestMessages,
+}) {
+  const { id } = useParams();
+  const history = useHistory();
+  const appointment = appointmentDetails?.[id];
+
+  useEffect(() => {
+    const fetchedPending = pendingStatus === FETCH_STATUS.succeeded;
+
+    if (!fetchedPending) {
+      history.push('/requested');
+    }
```

Comment:
For now, we are only allowing navigation to this page if coming from the "Requested" list.  If we haven't fetched pending, redirect back to the list.